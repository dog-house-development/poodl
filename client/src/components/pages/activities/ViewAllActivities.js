import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import _ from 'lodash';
import ActivityActions from '../../../actions/activityActions';
import DataGrid from '../../ui/DataGrid';
import { Link } from 'react-router-dom';
import moment from 'moment';

export class ViewAllActivities extends Component {
    constructor(props) {
        super(props);
        this.handleRowClick = this.handleRowClick.bind(this);
    }

    componentDidMount() {
        this.props.activityActions.filter();
    }

    handleRowClick(e, id) {
        e.preventDefault();
        this.props.history.push(`/activities/${id}`);
    }

    getDataGridContent() {
        let data = [];
        _.each(this.props.activities, activity => {
            data.push({
                name: activity.name,
                key: activity._id,
                startDate: moment(activity.startDate).format('h:mm a, MMMM Do YYYY'),
                endDate: moment(activity.endDate).format('h:mm a, MMMM Do YYYY')
            });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container page-container">
                <Link to="/dashboard" className="button small tertiary">
                    <i className="material-icons">keyboard_backspace</i> Back to home
                </Link>
                <div className="view-all-header">
                    <h1>Manage Activities</h1>
                    <Link to="/activities/add" className="button small primary">
                        <i className="material-icons button-icon">add</i>
                        Add Activity
                    </Link>
                </div>
                <DataGrid
                    data={this.getDataGridContent()}
                    loading={this.props.loading}
                    onRowClick={this.handleRowClick}
                />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        activities: state.activities.all,
        loading: state.activities.loading,
        errors: state.activities.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        activityActions: bindActionCreators(ActivityActions, dispatch)
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAllActivities);
