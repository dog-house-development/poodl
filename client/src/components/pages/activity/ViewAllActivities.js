import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import { fetchActivities } from '../../../actions/activityActions';
import DataGrid from '../../ui/DataGrid';
import { Link } from 'react-router-dom';
import moment from 'moment';

export class ViewAllActivities extends Component {
    constructor(props) {
        super(props);
        this.handleRowClick = this.handleRowClick.bind(this);
    }

    componentDidMount() {
        this.props.getActivities();
    }

    handleRowClick(e, id) {
        e.preventDefault();
        this.props.history.push(`/activity/${id}`);
    }

    getDataGridContent() {
        let data = [];
        _.each(this.props.activities, activity => {
            data.push({
                name: activity.name,
                key: activity._id,
                startDate: moment(activity.startDate).format('MMMM Do, YYYY'),
                endDate: moment(activity.endDate).format('MMMM Do, YYYY')
            });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container">
                <Link to="/dashboard" className="button small tertiary">
                    <i className="material-icons">keyboard_backspace</i> Back to home
                </Link>
                <div className="view-all-header">
                    <h1>Manage Activities</h1>
                    <Link to="/activities/add" className="button small primary">
                        <i className="material-icons button-icon">add</i>
                        Add Activity
                    </Link>
                </div>
                <DataGrid
                    data={this.getDataGridContent()}
                    loading={this.props.loading}
                    onRowClick={this.handleRowClick}
                />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        activities: state.activities.all,
        loading: state.activities.loading,
        errors: state.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        getActivities: () => dispatch(fetchActivities())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAllActivities);
