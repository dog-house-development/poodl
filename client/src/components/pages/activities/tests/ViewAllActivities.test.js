import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';
import { Link } from 'react-router-dom';

import { ViewAllActivities, mapStateToProps, mapDispatchToProps } from '../ViewAllActivities';

configure({ adapter: new Adapter() });

describe('View all activities tests', () => {
    let state, props, wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                activities: {
                    loading: false,
                    all: [
                        {
                            _id: '1',
                            name: 'Fun',
                            description: 'A fun time with fun activities',
                            startDate: '1971-02-01T07:00:00.000Z',
                            endDate: '1971-02-01T07:00:00.000Z',
                            members: [
                                { firstName: 'Big', lastName: 'Tup', _id: '123' },
                                { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                            ]
                        },
                        {
                            _id: '2',
                            name: 'Stuff',
                            description: 'A fun time with fun stuff',
                            startDate: '1970-01-01T07:00:00.000Z',
                            endDate: '1970-01-01T07:00:00.000Z',
                            members: [
                                { firstName: 'Big', lastName: 'Tup', _id: '123' },
                                { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                            ]
                        }
                    ],
                    errors: {}
                }
            },
            _state
        );
        props = _.assign({}, _props);
        wrapper = shallow(
            <ViewAllActivities
                {..._.assign(
                    {},
                    props,
                    mapStateToProps(state, props),
                    mapDispatchToProps(jasmine.createSpy('dispatch'))
                )}
            />
        );
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                activities: [
                    {
                        _id: '1',
                        name: 'Fun',
                        description: 'A fun time with fun activities',
                        startDate: '1971-02-01T07:00:00.000Z',
                        endDate: '1971-02-01T07:00:00.000Z',
                        members: [
                            { firstName: 'Big', lastName: 'Tup', _id: '123' },
                            { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                        ]
                    },
                    {
                        _id: '2',
                        name: 'Stuff',
                        description: 'A fun time with fun stuff',
                        startDate: '1970-01-01T07:00:00.000Z',
                        endDate: '1970-01-01T07:00:00.000Z',
                        members: [
                            { firstName: 'Big', lastName: 'Tup', _id: '123' },
                            { firstName: 'Lil', lastName: 'Tup', _id: '321' }
                        ]
                    }
                ],

                errors: {},
                loading: false
            });
        });
    });

    describe('mapDispatchToProps', () => {
        it('should map dispatch to props', () => {
            const dispatch = jest.fn();
            expect(JSON.stringify(mapDispatchToProps(dispatch))).toEqual(JSON.stringify({ activityActions: {} }));
        });
    });

    describe('componentDidMount', () => {
        it('should run without breaking', () => {
            spyOn(instance, 'componentDidMount');
            instance.componentDidMount();
        });
    });

    describe('getDataGridContent', () => {
        it('should return filtered activity data', () => {
            expect(instance.getDataGridContent()).toEqual([
                {
                    key: '1',
                    name: 'Fun',
                    startDate: '12:00 am, February 1st 1971',
                    endDate: '12:00 am, February 1st 1971',
                    memberCount: 2
                },
                {
                    key: '2',
                    name: 'Stuff',
                    startDate: '12:00 am, January 1st 1970',
                    endDate: '12:00 am, January 1st 1970',
                    memberCount: 2
                }
            ]);
        });
    });

    describe('render', () => {
        xit('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
