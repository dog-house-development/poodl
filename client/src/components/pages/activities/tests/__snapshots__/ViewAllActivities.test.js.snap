// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View all activities tests render should render correctly 1`] = `
<div
  className="view-all-container"
>
  <Link
    className="button small tertiary"
    replace={false}
    to="/dashboard"
  >
    <i
      className="material-icons"
    >
      keyboard_backspace
    </i>
     Back to home
  </Link>
  <div
    className="page-header"
  >
    <h1>
      Manage Activities
    </h1>
    <Link
      className="button small primary"
      replace={false}
      to="/activities/add"
    >
      <i
        className="material-icons button-icon"
      >
        add
      </i>
      Add Activity
    </Link>
  </div>
  <DataGrid
    data={
      Array [
        Object {
          "endDate": "11:00 pm, December 31st 1969",
          "key": "1",
          "name": "Fun",
          "startDate": "11:00 pm, December 31st 1969",
        },
        Object {
          "endDate": "11:00 pm, December 31st 1969",
          "key": "2",
          "name": "Stuff",
          "startDate": "11:00 pm, December 31st 1969",
        },
      ]
    }
    loading={false}
    onRowClick={[Function]}
  />
</div>
`;
